test/byContract/IexecEscrow/IexecEscrowNative.ts

```diff
@@ -0,0 +1,302 @@
+// SPDX-FileCopyrightText: 2020-2024 IEXEC BLOCKCHAIN TECH <contact@iex.ec>
+// SPDX-License-Identifier: Apache-2.0
+
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { BigNumber } from 'ethers';
+import { ethers, expect } from 'hardhat';
+import CONFIG from '../../../config/config.json';
+import { loadHardhatFixtureDeployment } from '../../../scripts/hardhat-fixture-deployer';
+import { IexecInterfaceNative, IexecInterfaceNative__factory } from '../../../typechain';
+import { getIexecAccounts } from '../../../utils/poco-tools';
+
+const DEPOSIT_AMOUNT = BigNumber.from(100);
+const DEPOSIT_AMOUNT_ETHER = ethers.utils.parseUnits(DEPOSIT_AMOUNT.toString(), 9);
+const ZERO_ADDRESS = ethers.constants.AddressZero;
+
+if (CONFIG.chains.default.asset === 'Native') {
+    describe('EscrowNative', () => {
+        let proxyAddress: string;
+        let [iexecPoco, iexecPocoAsAnyone]: IexecInterfaceNative[] = [];
+        let [iexecAdmin, anyone]: SignerWithAddress[] = [];
+
+        beforeEach('Deploy', async () => {
+            proxyAddress = await loadHardhatFixtureDeployment();
+            await loadFixture(initFixture);
+        });
+
+        async function initFixture() {
+            const accounts = await getIexecAccounts();
+            ({ iexecAdmin, anyone } = accounts);
+
+            iexecPoco = IexecInterfaceNative__factory.connect(proxyAddress, iexecAdmin);
+            iexecPocoAsAnyone = iexecPoco.connect(anyone);
+        }
+
+        async function checkBalances(address: string, expectedBalance: BigNumber) {
+            expect(await iexecPocoAsAnyone.balanceOf(address)).to.equal(expectedBalance);
+        }
+
+        describe('Receive and Fallback', () => {
+            it('Should call receive successfully', async () => {
+                await checkBalances(anyone.address, BigNumber.from(0));
+                const initialBalance = await ethers.provider.getBalance(anyone.address);
+
+                const tx = await anyone.sendTransaction({
+                    to: iexecPoco.address,
+                    value: DEPOSIT_AMOUNT_ETHER,
+                });
+                await expect(tx)
+                    .to.changeEtherBalances(
+                        [anyone, iexecPoco],
+                        [-DEPOSIT_AMOUNT_ETHER, DEPOSIT_AMOUNT_ETHER],
+                    )
+                    .to.emit(iexecPocoAsAnyone, 'Transfer')
+                    .withArgs(ZERO_ADDRESS, anyone.address, DEPOSIT_AMOUNT);
+
+                await checkBalances(anyone.address, DEPOSIT_AMOUNT);
```

Comment:
Why not using `changeTokenBalances` in previous `expect`?

Reply:
Thanks!

Reply:
I did not know about it

Reply:
Done